File: customer/src/main/java/br/com/accenture/wallet/customer/repository/CustomerRepository.java
package br.com.accenture.wallet.customer.repository;

import br.com.accenture.wallet.customer.domain.Customer;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface CustomerRepository extends JpaRepository<Customer, String> {

    @Query("SELECT c FROM Customer c WHERE c.email = :email")
    Optional<Customer> findByEmail(@Param("email") String email);

}

File: customer/src/main/kotlin/br/com/accenture/wallet/customer/repository/CustomerRepository.kt
package br.com.accenture.wallet.customer.repository

import br.com.accenture.wallet.customer.domain.Customer
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

@Repository
interface CustomerRepository : JpaRepository<Customer, String> {

    @Query("SELECT c FROM Customer c WHERE c.email = :email")
    fun findByEmail(@Param("email") email: String): Optional<Customer>

}
